Realm/RLMObjectBase.mm

```diff
@@ -544,7 +544,8 @@ void populateProperties(realm::CollectionChangeSet const& c) {
 
         auto properties = [NSMutableArray new];
         for (RLMProperty *property in object->_info->rlmObjectSchema.properties) {
-            if (c.columns.count(object->_info->tableColumn(property).value)) {
+            auto columnKey = object->_info->tableColumn(property).value;
```

Comment:
tableColumn retrieves the property index from `persisted_properties`
```
realm::ColKey RLMClassInfo::tableColumn(RLMProperty *property) const {
    return objectSchema->persisted_properties[property.index].column_key;
}
```
In the case of backlinks, the property index must be taken from `computed_properties`. How should this be done?
Should there be some branch within populateProperties that checks for whether this is called on a backlink?
Should there be a completely different method? A different impl.after()?

Reply:
What is the core side of things doing to indicate that a backlink property was modified? If it's adding the target table's ColKey to `columns` then you could just loop over computedProperties in the same way as this is looping over properties, but if it is doing that then we'd get incorrect results if both the source table and target table have columns with the same key...

Reply:
That's what happening in core. And [`9c20a19`](https://github.com/realm/realm-cocoa/pull/7249/commits/9c20a19a0db6685b1c1198eb22677080f2d3c068) loops over the computed properties. I suppose it _would_ be possible for the origin and target have the same column keys.
Would we have to scrap looping over computed_properties altogether? Or add an assert to make it not fail silently in when they do match? Or something else?